bridges/snowbridge/pallets/system-frontend/src/lib.rs

```diff
@@ -233,12 +310,46 @@ pub mod pallet {
 			T::XcmSender::deliver(ticket)
 		}
 
+		/// Swaps a specified tip asset to Ether and then burns the resulting ether for
+		/// teleportation. Returns the amount of Ether gained if successful, or a DispatchError if
+		/// any step fails.
+		fn swap_and_burn(
+			who: AccountIdOf<T>,
+			tip_asset_location: Location,
+			ether_location: Location,
+			tip_amount: u128,
+		) -> Result<u128, DispatchError>
+		where
+			<T as frame_system::Config>::AccountId: Into<Location>,
+		{
+			// Swap tip asset to ether
+			let swap_path = vec![tip_asset_location.clone(), ether_location.clone()];
+			let who_location: Location = who.clone().into();
+
+			let ether_gained = T::Swap::swap_exact_tokens_for_tokens(
+				who.clone(),
+				swap_path,
+				tip_amount,
+				None,
```

Comment:
I would add a comment here since this `None` is kind of obscure. `// We require no minimum amount` or something

Reply:
Added in https://github.com/paritytech/polkadot-sdk/pull/8271/commits/7f5249ed6d7031aacb28e1fd94d36e45d08da797.